### PR TITLE
Updated calibration method 

### LINKED ISSUE #104: Complete calibration part in runInit function of sciencelab
Complete runInit function of science lab to successfully read and process calibration data.
https://github.com/fossasia/pslab-android/blob/master/app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java#L108

After this we can check other functionality of other applications by logging and processing data applying calibration.

### LINKED ISSUE #121: Error in reading complete data from PSLab device
I tried reading 2048 bytes from Pslab device and was only able to read 1889 bytes. (was reading flash data from page 3 i.e ADC_POLYNOMIALS_LOCATION)
I tried many times but every time I was reading 1889 bytes after that error occurred.

screenshot from 2017-05-17 21-26-37

### LINKED ISSUE #129: Error in unpacking float data in sciencelab
The unpacking of float data from byte string read from PSLab is faulty [here](https://github.com/fossasia/pslab-android/blob/master/app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java#L154).
Sub-issue of https://github.com/fossasia/pslab-android/issues/104

### PR DESCRIPTION
**Pull request by viveksb007**
Fixes #121 Successfully read complete calibration data in chunks.
Fixes #129 Unpacked Correctly
Partly Fixes #104 ( creating a Polynomial Function from read values is remaining )
All the above issues were from same method, so updated them in single PR.

### CODE REVIEW

File: app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java [hunk 1]
```diff
@@ -167,17 +193,157 @@ private void runInitSequence(Boolean loadCalibrationData) throws IOException {
...
+                        e.printStackTrace();
+                        continue;
+                    }
+                    double slope = fits[0];
+                    double intercept = fits[1];
+                    double[] fitVals = Arrays.copyOfRange(fits, 2, fits.length);
+                    if ("PV1".equals(name) | "PV2".equals(name) | "PV3".equals(name)) {
+
+                        double[] DACX = new double[4096];
+                        double factor = (this.dac.chans.get(name).range[1] - this.dac.chans.get(name).range[0]) / 4096;
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java]: Hi @viveksb007 !
`this.dac.CHANS` has to be `this.dac.chans`

File: app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java [hunk 2]
```diff
@@ -167,17 +193,157 @@ private void runInitSequence(Boolean loadCalibrationData) throws IOException {
...
+                Map<String, ArrayList<Double[]>> polyDict = new LinkedHashMap<>();
+                String[] adcSlopeOffsetsSplit = adcSlopeOffsets.split(">|");
+                for (int i = 0; i < adcSlopeOffsetsSplit.length; i++) {
+                    if (i == 0) continue;
+                    String cals = adcSlopeOffsetsSplit[i].substring(5);
+                    polyDict.put(adcSlopeOffsetsSplit[i].substring(0, 3), new ArrayList<Double[]>());
+                    for (int j = 0; j < cals.length() / 16; j++) {
+                        Double[] poly = new Double[4];
+                        for (int k = 0; k < 4; k++) {
+                            poly[k] = ByteBuffer.wrap(cals.substring(k, k + 4).getBytes()).getDouble();
```
Comment (asitava1998) [inline app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java]: Please check this line. (k, k + 4) I think it should be (k +4*j, k + 4*(j+1)). Please correct me if I am wrong.

Comment (asitava1998) [inline app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java]: Please check this line. I think it should be (k + 4j, k + 4(j+1)). Please correct me if I am wrong.

Comment (viveksb007) [inline app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java]: I think it should be ((16 * j) + (k * 4), (16 * j) + (k * 4) + 4).

Comment (asitava1998) [inline app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java]: Yes, you are right. Hadn't looked at it properly

**Pull request merged**